### PR TITLE
new MergeIterator and Merge building function

### LINKED ISSUE #3: add MergeBuilder
The MergeBuilder would take n FST instances and a Writer as input, and produce a new FST containing the merged contents.  Implementation just has to create iterators on the underlying FSTs, walk them, and insert to the new one in order.

### PR DESCRIPTION
**Pull request by mschoch**
- fix builder, to copy last key, not use reference, which it doesn't own
- make Iterator an interface
- MergeIterator can iterate over several Iterators and invoke custom
  MergeFunc when the same key appears on multiple Iterators
- Merge function allows you to build new FST by merging several others

fixes #3

### CODE REVIEW

File: merge_iterator.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+func (m *MergeIterator) init() {
+	for i, itr := range m.itrs {
+		m.currKs[i], m.currVs[i] = itr.Current()
+	}
+	m.updateMatches()
+}
+
+func (m *MergeIterator) updateMatches() {
+	m.lowK = m.currKs[0]
```
Comment (steveyen) [inline merge_iterator.go]: maybe one edge case is when len(currKs) is 0?

**Comment by mschoch** [comment]
Thanks Steve, I added a test case for this scenario, and made the behavior for that case a bit more friendly.

**Comment by seiflotfy** [comment]
great work @mschoch 👍
